src/v/storage/key_offset_map.h

```diff
@@ -179,7 +179,8 @@ class hash_key_offset_map : public key_offset_map {
     hash_type::digest_type hash_key(const compaction_key&) const;
 
     mutable hash_type hasher_;
-    large_fragment_vector<entry> entries_;
+    fragmented_vector<entry, 64 * 1024> entries_;
```

Comment:
You could consider using chunked_vector that will automatically calculate the fragment size too

Reply:
chunked_vector provides a fragment size of 80KB for this case, so that would work. they both suffer the same issue when the offset map target size increases sufficiently. is there any advantage to switching? afaict chunk_vector helps when the first fragment may be small. this container will always have something like 128MB total.

Reply:
That is fair that the outer vector is so large in this case it overallocs. One way to fix that would be to write our own inner vector type, as we can save 8 bytes as we track the capacity separately. Honestly maybe even the size could be dropped as we track that too in aggregate and could compute the single fragment size when needed (which means we could just store an array of pointers).

The automatic size computation of chunked_vector seems superior to manually picking a fragment size, as I see no downsides to 80k vs 64k chunks (should improve the fragmentation minorly actually)

Reply:
>superior to manually picking a fragment size

yeh good point! switched to chunked_vector